frame/bounties/src/lib.rs

```diff
@@ -660,12 +785,686 @@ decl_module! {
+
+					// update the subbounty state
+					subbounty.status = SubBountyStatus::SubCuratorProposed {
+						subcurator: subcurator
+					};
+					Ok(())
+				}
+			)?;
+		}
+
+		/// Accept the subcurator role for the subbounty.
+		///
+		/// The dispatch origin for this call must be
+		/// the subcurator of this subbounty.
+		///
+		/// A deposit will be reserved from the subcurator and
+		/// refund upon successful payout or cancellation.
+		///
+		/// Fee for subcurator is deducted from curator
+		/// fee of parent bounty.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in "SubCuratorProposed" state, for
+		/// processing the call. And state of subbounty is
+		/// moved to "Active" on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = <T as Config>::WeightInfo::accept_subcurator()]
+		fn accept_subcurator(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let signer = ensure_signed(origin)?;
+
+			// Ensure parent bounty exist
+			let _ = Self::ensure_bounty_exist(bounty_id, false)?;
+
+			// Mutate Subbounty
+			SubBounties::<T>::try_mutate_exists(bounty_id, subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure subbounty is in expected state
+				if let SubBountyStatus::SubCuratorProposed { ref subcurator } = subbounty.status
+				{
+					ensure!(signer == *subcurator, Error::<T>::RequireSubCurator);
+
+					// Reserve subcurator deposit
+					let deposit = T::BountyCuratorDeposit::get() * subbounty.fee;
+					T::Currency::reserve(subcurator, deposit)?;
+					subbounty.curator_deposit = deposit;
+
+					subbounty.status = SubBountyStatus::Active {
+						subcurator: subcurator.clone(),
+					};
+					Ok(())
+				} else {
+					Err(Error::<T>::UnexpectedStatus.into())
+				}
+			})?;
+		}
+
+		/// Unassign subcurator from a subbounty.
+		///
+		/// The dispatch origin for this call can be
+		/// either `RejectOrigin` or any signed origin.
+		///
+		/// For the origin other than T::RejectOrigin,
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work. For origin
+		/// T::RejectOrigin execution is forced by ignoring
+		/// the state of parent bounty.
+		///
+		/// If this function is called by the `RejectOrigin` or
+		/// the master curator, we assume that the subcurator is
+		/// malicious or inactive.
+		///
+		/// As a result, subcurator deposit may be slashed.
+		///
+		/// If the origin is the subcurator, we take this as a sign they are
+		/// unable to do their job, and they willingly give up.
+		/// We could slash the deposit, but for now we allow them to
+		/// unreserve their deposit and exit without issue.
+		/// (We may want to change this if it is abused.)
```

Comment:
I would also consider splitting this dispatchable into smaller ones, as it is quite complicated with regard to origin. Something like: 

- `unassign_subcurator_own`
- `unassign_subcurator_forced`

Reply:
Indeed, I agree its quite big block and can be refactored. And wish to take it as separate PR